scheduler/reconcile.go

```diff
@@ -201,29 +212,28 @@ func (a *allocReconciler) Compute() *reconcileResults {
 	}
 
 	// Detect if the deployment is paused
-	if a.deployment != nil {
-		a.deploymentPaused = a.deployment.Status == structs.DeploymentStatusPaused ||
-			a.deployment.Status == structs.DeploymentStatusPending
-		a.deploymentFailed = a.deployment.Status == structs.DeploymentStatusFailed
-	}
-	if a.deployment == nil {
-		// When we create the deployment later, it will be in a pending
-		// state. But we also need to tell Compute we're paused, otherwise we
-		// make placements on the paused deployment.
-		if a.job.IsMultiregion() && !(a.job.IsPeriodic() || a.job.IsParameterized()) {
-			a.deploymentPaused = true
-		}
-	}
+	a.computePaused()
```

Comment:
#Refactoring review

Reply:
This should probably be `a.computeDeploymentPaused()`, right? I'd also remove the useless comment left above this line because the name says what it does.

This refactoring is fine. But absolutely pull this out to its own PR.

Reply:
Will do. FYI, this is the kind of refactoring I stopped myself from doing throughout all of `computeGroup`.

Reply:
The scheduler code is full of spaghetti. Rabbit holing on refactoring on your first trip through it will end in tears. 😀